crates/bevy_window/src/window.rs

```diff
@@ -624,23 +624,50 @@ impl Window {
     }
 }
 
+/// Describes the information needed for creating a window.
+///
+/// This should be set up before adding the [`WindowPlugin`](crate::WindowPlugin).
+/// Most of these settings can also later be configured through the [`Window`](crate::Window) resource.
+///
+/// See [`examples/window/window_settings.rs`] for usage.
+///
+/// [`examples/window/window_settings.rs`]: https://github.com/bevyengine/bevy/blob/latest/examples/window/window_settings.rs
 #[derive(Debug, Clone)]
 pub struct WindowDescriptor {
+    /// The requested logical width of the window's client area.
+    /// May vary from the physical width due to different pixel density on different monitors.
     pub width: f32,
+    /// The requested logical height of the window's client area.
+    /// May vary from the physical height due to different pixel density on different monitors.
     pub height: f32,
+    /// The position on the screen that the window will be centered at.
+    /// If set to `None`, some platform-specific position will be chosen.
     pub position: Option<Vec2>,
+    /// Sets minimum and maximum resize limits.
     pub resize_constraints: WindowResizeConstraints,
+    /// Overrides the window's ratio of physical pixels to logical pixels.
     pub scale_factor_override: Option<f64>,
```

Comment:
So after some discussion on discord, we discovered that on X11 the default scale factor is a bit weird and doesn't always defaults to 1.0.

See the thread here https://discord.com/channels/691052431525675048/977977143004905584/978161092155031552

The fix was to force a scale factor override of 1.0. Just reading the winit docs makes it pretty clear that this is a complicated issue for X11 https://docs.rs/winit/latest/winit/dpi/index.html#how-is-the-scale-factor-calculated

I think it would be a good idea to at least mention here that if users are having unexpected scaling when using X11 they should try setting it to Some(1.0)